src/mixins/java/org/spongepowered/common/mixin/tracker/network/play/ServerPlayNetHandlerMixin_Tracker.java

```diff
@@ -155,4 +175,40 @@
         }
         SpongeCommonEventFactory.lastPrimaryPacketTick = SpongeCommon.getServer().getTickCounter();
     }
+
+    @Inject(method = "processUseEntity", cancellable = true,
+            at = @At(value = "INVOKE",
+                    target = "Lnet/minecraft/entity/player/ServerPlayerEntity;interactOn(Lnet/minecraft/entity/Entity;Lnet/minecraft/util/Hand;)Lnet/minecraft/util/ActionResultType;"),
+            locals = LocalCapture.CAPTURE_FAILHARD
+    )
+    public void onRightClickEntity(CUseEntityPacket packetIn, CallbackInfo ci, ServerWorld serverworld, Entity entity) {
+        InteractEntityEvent.Secondary event = SpongeCommonEventFactory.callInteractEntityEventSecondary(this.player, this.player.getHeldItem(packetIn.getHand()), entity, packetIn.getHand(), null);
```

Comment:
Category "What do you think I will say?" for $200 Alex.

Reply:
@dualspiral

Reply:
How does @Zidane feel about the `final` things?